pkg/machines/components/networks/createNetworkDialog.jsx

```diff
@@ -403,10 +403,16 @@ class CreateNetworkModal extends React.Component {
             const ipv6 = ip === "IPv4 only" ? undefined : this.state.ipv6;
             const ipv4 = ip === "IPv6 only" ? undefined : this.state.ipv4;
             const netmask = utils.netmaskConvert(this.state.netmask);
+            // If you take an IP address an apply the network mask (bitwise and) then you get the network address - no host can have such address
+            // Increment by 1 to get the first usable address of the block
+            const ipv4FirstUsableAddress = (ipv4 &&
+                ipv4.split(".")
+                        .map((item, index) => (item & netmask.split(".")[index]) + Number(index == 3))
+                        .join("."));
 
             this.setState({ createInProgress: true });
             networkCreate({
-                connectionName, name, forwardMode, device, ipv4, netmask, ipv6, prefix,
+                connectionName, name, forwardMode, device, ipv4: ipv4FirstUsableAddress, netmask, ipv6, prefix,
```

Comment:
Let me understand this properly, what this code does is that it simply takes ANY ip address and user inputted in IPv4 Network input field and increases it by one?

Shouldn't we change the default address in the UI to "192.168.100.1" so what user sees in UI corresponds with what's then put into XML?

Also what would happen if user for whatever reason put address which ends with .255. Would it just increase by one and jump out of valid IPv6 range?

Reply:
That's what virt-manager does - just increases by one I think. (playing around with the UI at least gave me this impression)
Good point for the address which ends with .255. We should not allow that I guess.

Reply:
To me it seems that virt-manager doesn't increase it by one. To me it seems it ignores whatever you put into last part of IPv4 address and just replaces it with .1
Example: in virt-manager, I tried to create virtual network with 192.168.100.77/24, and virt-manager put 192.168.100.1 into XML.
So to me it seems it just takes the part defined by submask from user input field (the first 24 bits in this case) and overwrites rest with 1